### PR TITLE
get shell integration to work for zsh

### LINKED ISSUE #141620: Shell integration zsh hooks don't load on macOS/Linux sometimes
Testing #141339

I get this error in the terminal, on Linux using zsh

```
/snap/code-insiders/917/usr/share/code-insiders/resources/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-zsh.sh: 50: /snap/code-insiders/917/usr/share/code-insiders/resources/app/out/vs/workbench/contrib/terminal/browser/media/shellIntegration-zsh.sh: Syntax error: word unexpected (expecting ")")
```

```
$ zsh --version
zsh --version
zsh 5.4.2 (x86_64-ubuntu-linux-gnu)
```

### PR DESCRIPTION
**Pull request by meganrogge**
This PR fixes #141620

### CODE REVIEW

File: src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts [hunk 1]
```diff
@@ -273,6 +273,10 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 							shellLaunchConfig.env = shellLaunchConfig.env || {} as IProcessEnvironment;
 							shellLaunchConfig.env['VSCODE_SHELL_LOGIN'] = '1';
 						}
+						if (env?.['ZDOTDIR']) {
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts:276]: I think `?.` for here and above isn't actually needed anymore, we should also remove the TODO: fix above

Comment (Tyriar) [reply src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts:276]: We may also be able to remove these all together as env is passed into `createProcess`? See how `VSCODE_SHELL_HIDE_WELCOME` doesn't need to get set on `shellLaunchConfig.env`

File: src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts [hunk 2]
```diff
@@ -492,6 +492,7 @@ export function injectShellIntegrationArgs(
 				} else if (areZshBashLoginArgs(originalArgs)) {
 					newArgs = shellIntegrationArgs.get(ShellIntegrationExecutable.ZshLogin);
 				}
+				env['ZDOTDIR'] = '${execInstallFolder}/out/vs/workbench/contrib/terminal/browser/media/';
```
Comment (Tyriar) [inline src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts:495]: Let's explain in a comment what's happening and why this approach was taken

**Review by Tyriar — APPROVED** [review]

File: src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts (hunk 1 above)
Comment (meganrogge) [reply src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts:276]: we need to resolve the execInstallFolder there for the ZDOTDIR one
and see this TODO - slc.env is used, not env in remote terminal backend https://github.com/microsoft/vscode/blob/4a130c40ed876644ed8af2943809d08221375408/src/vs/workbench/contrib/terminal/browser/remoteTerminalBackend.ts#L161

**Pull request merged**